pkg/repository/v1/payloadstore.go

```diff
@@ -0,0 +1,450 @@
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/hatchet-dev/hatchet/internal/telemetry"
+	"github.com/hatchet-dev/hatchet/pkg/repository/postgres/sqlchelpers"
+	"github.com/hatchet-dev/hatchet/pkg/repository/v1/sqlcv1"
+	"github.com/jackc/pgx/v5/pgtype"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/rs/zerolog"
+)
+
+type StorePayloadOpts struct {
+	Id         int64
+	InsertedAt pgtype.Timestamptz
+	Type       sqlcv1.V1PayloadType
+	Payload    []byte
+}
+
+type RetrievePayloadOpts struct {
+	Id         int64
+	InsertedAt pgtype.Timestamptz
+	Type       sqlcv1.V1PayloadType
+}
+
+type PayloadLocation string
+type ExternalPayloadLocationKey string
+
+const (
+	PayloadLocationInline   PayloadLocation = "inline"
+	PayloadLocationExternal PayloadLocation = "external"
+)
+
+type PayloadContent struct {
+	Location            PayloadLocation             `json:"location"`
+	ExternalLocationKey *ExternalPayloadLocationKey `json:"external_location_key,omitempty"`
+	InlineContent       []byte                      `json:"inline_content,omitempty"`
+}
+
+type ExternalStore interface {
+	Store(ctx context.Context, tenantId string, payloads ...StorePayloadOpts) (map[RetrievePayloadOpts]ExternalPayloadLocationKey, error)
+	BulkRetrieve(ctx context.Context, tenantId string, keys ...ExternalPayloadLocationKey) (map[ExternalPayloadLocationKey][]byte, error)
+}
+
+type PayloadStoreOption func(*payloadStoreRepositoryImpl)
+
+func WithExternalStore(store ExternalStore, externalStoreLocationName string, nativeStoreTTL time.Duration) PayloadStoreOption {
```

Comment:
Instead of `WithExternalStore` here, we could add `OverwriteExternalStore` as a method to the `PayloadStoreRepository`, which allows us to overwrite the external store after we've initialized the repository which is what we want.